### PR TITLE
add TSMappedType and TSTypeParameter

### PR DESCRIPTION
**Pull request by despairblue**

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -1974,6 +1974,25 @@ function genericPrintNoParens(path, options, print, args) {
         ") => ",
         path.call(print, "typeAnnotation"),
       ])
+    case "TSMappedType":
+      return concat([
+        "{ [",
```
Comment (vjeux) [inline src/printer.js]: the space here needs to be optional based on the `bracketSpacing` option

**Comment by vjeux** [comment]
Looks like it conflicts with master :( If you rebase I'll merge it.

**Comment by despairblue** [comment]
rebased and added the bracketSpacing

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**